Lib/sqlite3/dbapi2.py

```diff
@@ -55,16 +55,25 @@ def TimestampFromTicks(ticks):
 collections.abc.Sequence.register(Row)
 
 def register_adapters_and_converters():
+    from warnings import warn
+
+    msg = ("The default {what} are deprecated as of Python 3.12; "
+           "see the sqlite3 documentation for suggested replacement recipes")
+
     def adapt_date(val):
+        warn(msg.format(what="adapters"), DeprecationWarning, stacklevel=2)
         return val.isoformat()
 
     def adapt_datetime(val):
+        warn(msg.format(what="adapters"), DeprecationWarning, stacklevel=2)
         return val.isoformat(" ")
 
     def convert_date(val):
+        warn(msg.format(what="converters"), DeprecationWarning, stacklevel=2)
         return datetime.date(*map(int, val.split(b"-")))
 
     def convert_timestamp(val):
+        warn(msg.format(what="converters"), DeprecationWarning, stacklevel=2)
```

Comment:
Can use a slightly more descriptive `what`?, e.g.
- `date adapter`
- `datetime adapter`
- `date converter`
- `datetime converter`

Reply:
Thanks! Addressed in 6213c3bb2628ed5b9826cd6104e81b595f3b65ea